analysis/fake_data_stan_analysis.py

```diff
@@ -0,0 +1,391 @@
+    rReaderProcessor.Run()
+    analysisProcessor.data = rReaderProcessor.data
+
+    #Run analysis
+    analysisProcessor.Run()
+    results = analysisProcessor.results
+    
+    #Save results
+    writerProcessor.data = results
+    writerProcessor.Run()
+    
+    return results
+
+
+def PlotStanResults(posteriors, correlation_vars=['Q', 'KEmin', 'lp_prob'], divergence_vars=['Q', 'sigma', 'mass']):
+    """
+    Creates and saves two plots:
+        a) posteriors and correlations between them, and
+        b) plot showing where in parameter space Hamiltonian Monte Carlo divergences occured.
+        
+    Required argument:
+        1) posteriors: dict; output of the PyStanSamplingProcessor
+    Optional arguments:
+        2) correlation_vars: list of strings; names of variables to be included in correlation plot
+        3) divergence_vars: list of strings; names of variables to be included in divergences plot
+    """
+    aposteriori_config = {
+        "n_bins_x": 50,
+        "n_bins_y": 50,
+        "variables": correlation_vars,
+        "title": "Q_vs_Kmin",
+        "output_path": "./plots"
+    }
+    
+    div_plot_config = {
+        "n_bins_x": 50,
+        "n_bins_y": 50,
+        "variables": divergence_vars,
+        "title": "div_plot",
+        "output_path": "./plots"
+    }
+    
+    #Configuration step
+    aposterioriPlotter.Configure(aposteriori_config)
+    divPlotter.Configure(div_plot_config)
+    
+    #Plots correlations between posteriors
+    aposterioriPlotter.data = posteriors
+    aposterioriPlotter.Run()
+    
+    #Plot 2D grid of divergent points
+    divPlotter.data = posteriors
+    divPlotter.Run()
+
+
+def PerformFakeExperiment(root_filename, plot_results=False, parallelized=True):
+    """
+    Generate fake tritium data and analyze it in Stan. If plot_results==True, create correlation and divergence plots.
+    
+    Saves generation inputs, generation results, and analysis results to different trees of one ROOT file.
+    """
+    if parallelized==True:
+        flist = root_filename.split('/')
+        logger.info("----------MORPHO RUN #{}----------".format(flist[len(flist)-1][0]))
+    
+    #Sample inputs to the data generator and save to one branch of a root file:
+    inputs_dict = DefineGeneratorInputs(root_filename)
+    
+    #If fixing all inputs:
+    #inputs_dict = {"Q":QT2(), "mass":0.2, "KEmin":16323, "KEmax":19573.24, "sigma":18.6, "S":3300, "B_1kev":0.1, "scattering_prob":0.77, "err_from_B":0.5, "Nscatters":20}
+    
+    #Generate data using the inputs and save data
+    tritium_data = GenerateFakeData(inputs_dict, root_filename)
+    
+    #Analyze data and save posteriors
+    posteriors = StanTritiumAnalysis(tritium_data, root_file=root_filename, F_or_K='K')
+    
+    #Optionally plot posteriors
+    if plot_results == True:
+        PlotStanResults(posteriors)
+        
+    #Dictionary containing Stan convergence diagnostics information
```

Comment:
This does not need to be added before we run the analysis this week. 

If this becomes an "example" script in mermithid, it could serve as a template for future fake data study workflows. In that case, keeping track of diagnostics from all runs is a good feature to have, so I'll plan to add it in at some future point.